### PR TITLE
Fix sequence expand op

### PR DESCRIPTION
**Pull request by ktlichkid**
Sequence expand op's GPU grad kernel implementation is not robust enough if memory optimizer is on. 

The GPU kernel directly computed the sum of gradient without checking the initial value in d_x tensor.

In this PR, I moved the "set zero" function outside the functor to guarantee d_x is set to zero both on CPU and GPU.

### CODE REVIEW

File: paddle/fluid/operators/sequence_expand_op.h [hunk 1]
```diff
@@ -151,8 +151,8 @@ struct SequenceExpandGradFunctor<platform::CPUDeviceContext, T> {
       const framework::Vector<size_t>& x_lod,   /*expand source lod*/
       const framework::Vector<size_t>& ref_lod, /*expand referenced lod*/
       LoDTensor* dx) {
-    math::SetConstant<platform::CPUDeviceContext, T> set_zero;
-    set_zero(context, dx, static_cast<T>(0));
+    // math::SetConstant<platform::CPUDeviceContext, T> set_zero;
```
Comment (kuke) [inline paddle/fluid/operators/sequence_expand_op.h]: Please remove these two lines

Comment (ktlichkid) [reply paddle/fluid/operators/sequence_expand_op.h]: Done

**Review by wanghaoshuang — APPROVED** [review]
Great job! Actually, the `sequence expand op` may also give wrong gradient value even memory optimizer is off.

**Comment by kuke** [comment]
@wanghaoshuang Please have a test on the attention-based OCR model to make sure that this change  solves the problem.

**Review by reyoung — APPROVED** [review]
Excellent! thanks!

**Pull request merged**